### PR TITLE
Specify the docker tag during the build

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes the image tagging, currently failing on Travis: 
https://travis-ci.org/uber/jaeger/jobs/236245547#L1096

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -117,7 +118,7 @@ docker: build_ui build-agent-linux build-collector-linux build-query-linux
 	cp -r jaeger-ui-build/build/ cmd/query/jaeger-ui-build
 	docker build -t $(DOCKER_NAMESPACE)/jaeger-cassandra-schema plugin/storage/cassandra/ ; \
 	for component in agent collector query ; do \
-		docker build -t $(DOCKER_NAMESPACE)/jaeger-$$component cmd/$$component ; \
+		docker build -t $(DOCKER_NAMESPACE)/jaeger-$$component:${DOCKER_TAG} cmd/$$component ; \
```
Comment (black-adder) [inline Makefile]: We probably want to tag with "latest" too

Comment (jpkrohling) [inline Makefile]: That part is done on the `upload-to-docker.sh`, for builds running on `master`:
https://github.com/uber/jaeger/blob/master/travis/upload-to-docker.sh#L5

Comment (jpkrohling) [inline Makefile]: https://github.com/uber/jaeger/blob/master/travis/upload-to-docker.sh#L5

File: Makefile [hunk 2]
```diff
@@ -117,7 +118,7 @@ docker: build_ui build-agent-linux build-collector-linux build-query-linux
 	cp -r jaeger-ui-build/build/ cmd/query/jaeger-ui-build
 	docker build -t $(DOCKER_NAMESPACE)/jaeger-cassandra-schema plugin/storage/cassandra/ ; \
```
Comment (black-adder) [inline Makefile]: Do we need to tag this too then?

Comment (jpkrohling) [inline Makefile]: Fixed

**Pull request merged**